gateways/core/stream/amqp/start.go

```diff
@@ -44,6 +45,31 @@ func (ese *AMQPEventSourceExecutor) StartEventSource(eventSource *gateways.Event
 	return gateways.HandleEventsFromEventSource(eventSource.Name, eventStream, dataCh, errorCh, doneCh, ese.Log)
 }
 
+func getLimitedDelivery(ch *amqplib.Channel, a *amqp, delivery chan amqplib.Delivery, queue string) {
+	for {
+		var elapsedTime int64
+		startTime := time.Now().Unix()
+
+		for i := uint32(0); i < a.RateLimit; i++ {
```

Comment:
So If I get it right, if I send 100 messages, with a rate limit of 10msg,
I'll get 10msg at `00:00:00`, nothing for 59s, then another 10msg at `00:01:00`, etc, rather than a message at `00:00:00`, another one at `00:00:06`, etc

Am I correct ?

Reply:
I feel like this is not wrong, but with such scenario, I'll be better to have an extra configuration option to set the time window instead of the default 60s

Reply:
To avoid this, I introduced a new option `ratePeriod` which allow the user to set its own time window.